### PR TITLE
Don't include intercept by in regression methods by default.

### PR DESCRIPTION
**Pull request by cseed**
I will prepare a dev post describing the change.

### CODE REVIEW

**Comment by jbloom22** [comment]
cool, put the post in this tread:
http://discuss.hail.is/t/log-of-breaking-changes-in-0-2-beta/454

File: python/hail/methods/statgen.py [hunk 1]
```diff
@@ -281,22 +281,22 @@ def linear_regression(y, x, covariates=(), root='linreg', block_size=16) -> Matr
 
     .. math::
 
-        \mathrm{height} = \\beta_0 + \\beta_1 \, \mathrm{genotype}
-                          + \\beta_2 \, \mathrm{age}
-                          + \\beta_3 \, \mathrm{is_female}
+        \mathrm{height} = \\beta_0 \, \mathrm{genotype}
```
Comment (jbloom22) [inline python/hail/methods/statgen.py]: There still is an intercept in the example. In my lmm branch, I've moved to using beta_\star for the effect of interest, e.g.:
```
\\beta_\star \, \mathrm{genotype} + \\beta_0 + \\beta_1 \, \mathrm{age} + \\beta_1 \, \mathrm{is_female}
```
I think this is cleaner and clearer, could you do the same in linreg and logreg equation/table (lmmreg and skat don't need it).

You could also replace all the`\\` in linreg with `\` if you put make the docstring raw, or leave it.

File: python/hail/methods/statgen.py [hunk 2]
```diff
@@ -410,17 +412,17 @@ def logistic_regression(test, y, x, covariates=(), root='logreg') -> MatrixTable
     .. math::
 
         \mathrm{Prob}(\mathrm{is_case}) =
-            \mathrm{sigmoid}(\beta_0 + \beta_1 \, \mathrm{gt}
-                            + \beta_2 \, \mathrm{age}
-                            + \beta_3 \, \mathrm{is_female} + \varepsilon),
+            \mathrm{sigmoid}(\beta_0 \, \mathrm{gt}
```
Comment (jbloom22) [inline python/hail/methods/statgen.py]: you lost the intercept here too (see comment above)

File: src/main/scala/is/hail/methods/LinearRegression.scala [hunk 3]
```diff
@@ -31,14 +31,26 @@ object LinearRegression {
...
 
     if (d < 1)
-      fatal(s"$n samples and ${ k + 1 } ${ plural(k, "covariate") } (including x and intercept) implies $d degrees of freedom.")
+      fatal(s"$n samples and ${ k + 1 } ${ plural(k, "covariate") } (including x) implies $d degrees of freedom.")
 
     info(s"linear_regression: running on $n samples for ${ y.cols } response ${ plural(y.cols, "variable") } y,\n"
-       + s"    with input variable x, intercept, and ${ k - 1 } additional ${ plural(k - 1, "covariate") }...")
+       + s"    with input variable x, and ${ k } additional ${ plural(k, "covariate") }...")
+
+    println("y.shape", y.rows, y.cols)
```
Comment (jbloom22) [inline src/main/scala/is/hail/methods/LinearRegression.scala]: you left in four print statements

**Comment by tpoterba** [comment]
@jbloom22 can you take over this PR and make the changes?

**Comment by jbloom22** [comment]
incorporated into #4067

**Pull request closed without merging**